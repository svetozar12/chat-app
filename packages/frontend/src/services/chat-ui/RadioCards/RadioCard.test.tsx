import { Box, Heading, ScaleFade } from '@chakra-ui/react';
import { mount, ReactWrapper } from 'enzyme';
import RadioCard from 'services/chat-ui/RadioCards/RadioCards';
const componentProps = {
  closeRadioCard: () => console.log('test func'),
  heading: 'RadioCard Heading',
  children: <h1>test text</h1>,
  style: { background: 'lime' },
  chakraProps: { h: '20%' },
};

const baseProps = {
  background: 'lime',
  h: '20%',
};

const setupMount = () => {
  return mount(<RadioCard {...componentProps} />);
};

describe('testing correct behaviour of <RadioCard />', () => {
  let wrapper: ReactWrapper;
  beforeAll(() => {
    wrapper = setupMount();
  });
  it('component <RadioCard /> should render', () => {
    expect(wrapper.find(RadioCard).length).toBe(1);
  });
  Object.entries(baseProps).forEach(([key, value]) => {
    it(`component <RadioCard /> should have prop ${key}:${value}`, () => {
      expect(wrapper.find(RadioCard).children().prop(key)).toBe(value);
    });
  });
  it('component <RadioCard /> should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
