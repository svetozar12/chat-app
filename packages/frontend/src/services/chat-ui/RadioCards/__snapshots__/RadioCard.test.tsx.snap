// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing correct behaviour of <RadioCard /> component <RadioCard /> should match snapshot 1`] = `
<RadioCard
  chakraProps={
    Object {
      "h": "20%",
    }
  }
  closeRadioCard={[Function]}
  heading="RadioCard Heading"
  style={
    Object {
      "background": "lime",
    }
  }
>
  <Box
    as="label"
    background="lime"
    h="20%"
  >
    <Noop />
    <label
      className="css-0"
    >
      <input
        checked={false}
        id="radio-:r0:"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "border": "0px",
            "clip": "rect(0px, 0px, 0px, 0px)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": "0px",
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "1px",
          }
        }
        type="radio"
      />
      <Box
        _checked={
          Object {
            "bg": "#3182ce",
            "borderColor": "#3182ce",
            "color": "white",
          }
        }
        aria-hidden={true}
        borderRadius="md"
        borderWidth="1px"
        boxShadow="md"
        cursor="pointer"
        fontWeight="semibold"
        minW="6.25rem"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        px={5}
        py={3}
        textAlign="center"
      >
        <Noop />
        <div
          aria-hidden={true}
          className="css-0"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
        >
          <h1>
            test text
          </h1>
        </div>
      </Box>
    </label>
  </Box>
</RadioCard>
`;
